binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
```

Comment:
Ya so. We could order the keys. What do you mean by c-style? The driving motivation to do this was so the account permissions could be human readable in the genesis file, in arbitrary order, and could be displayed by the rpc. but it suggests that maybe binary could support a second "unsafe" mode, simply for human encoding convenience. Or else anything that ends up in a tree knows it cant have a map in its struct. Though now that I think of it, I could just use a json struct with the permissions and enforce the order with useful error msg. Maybe that's the better answer for now, and if there's a more burning desire for maps in `binary` we can revisit